src/party_one.rs

```diff
@@ -136,14 +136,36 @@ pub struct EphKeyGenSecondMsg {}
 //****************** End: Party One structs ******************//
 
 impl KeyGenFirstMsg {
-    pub fn create_commitments() -> (KeyGenFirstMsg, CommWitness, EcKeyPair) {
-        let base: GE = ECPoint::generator();
 
-        let secret_share: FE = ECScalar::new_random();
-        //in Lindell's protocol range proof works only for x1<q/3
-        let secret_share: FE =
-            ECScalar::from(&secret_share.to_big_int().div_floor(&BigInt::from(3)));
+    //in Lindell's protocol range proof works only for x1 \in {q/3 , ... , 2q/3}
+    pub fn is_secret_share_in_range(secret_share: &FE) -> bool {
+        let lower_bound: BigInt = FE::q().div_floor(&BigInt::from(3));
+        let upper_bound: BigInt = FE::q().mul(&BigInt::from(2))
+            .div_floor(&BigInt::from(3));
+
+        return if secret_share.to_big_int().gt(&lower_bound) &&
+            secret_share.to_big_int().lt(&upper_bound) {
+            true
+        } else {
+            false
+        }
+    }
+
+    pub fn get_secret_share_in_range() -> FE {
+        let mut secret_share: FE = ECScalar::new_random();
+        secret_share = ECScalar::from(&secret_share.to_big_int());
```

Comment:
Can be done in one line:             `secret_share = ECScalar::from(&ECScalar::new_random().to_big_int());`